gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/util/TraversalHelper.java

```diff
@@ -57,6 +60,23 @@
     private TraversalHelper() {
     }
 
+    public static boolean isNotBeyondElementId(final Traversal.Admin<?, ?> traversal) {
```

Comment:
We really need to get this method hidden as if you note me response. I don't think we have all case covered. This really should be in 3.2.0 when we have a slimmed down instruction set. What I will do on merge is move this to ComputerVerifcationStrategy and make it private. Then come 3.2.0 we can move it to TraversalHelper.